packages/console/components/DataTable.svelte

```diff
@@ -0,0 +1,487 @@
+		currentPage = pageNumber;
+		if (inputPage) {
+			inputPage = currentPage;
+		}
+	};
+
+	$: paginated = totalRows > rows;
+
+	$: if (paginated) {
+		pageCount = Math.ceil(totalRows / rows);
+	} else {
+		currentPage = 1;
+	}
+
+	let hovering = false;
+</script>
+
+{#if error === undefined}
+	<slot />
+	<div
+		class="table-container"
+		transition:slide|local
+		style="margin-top: 1.5em; margin-bottom: 1em; padding-bottom: 0em;"
+		on:mouseenter={() => (hovering = true)}
+		on:mouseleave={() => (hovering = false)}
+	>
+		<div class="container">
+			<table>
+				<thead>
+					<tr>
+						{#each columnSummary.filter((d) => d.show === true) as column}
+							<th class={column.type}>
+								<span class="col-header">
+									{column.title}
+								</span>
+							</th>
+						{/each}
+					</tr>
+				</thead>
+
+				{#each data as row (row)}
+					<tr in:blur>
+						{#each columnSummary as column}
+							<td class="{column.type} row-lines">
+								{formatValue(row[column.id], column.format, column.columnUnitSummary)}
+							</td>
+						{/each}
+					</tr>
+				{/each}
+			</table>
+		</div>
+
+		{#if paginated && pageCount > 1}
+			<div class="pagination">
+				<div class="page-labels">
+					<button
+						aria-label="first-page"
+						class="page-changer"
+						class:hovering
+						disabled={currentPage === 1}
+						on:click={() => goToPage(1)}
+					>
+						<div class="page-icon">
+							<MdFirstPage />
+						</div>
+					</button>
+					<button
+						aria-label="previous-page"
+						class="page-changer"
+						class:hovering
+						disabled={currentPage === 1}
+						on:click={() => goToPage(currentPage - 1)}
+					>
+						<div class="page-icon">
+							<MdNavigateBefore />
+						</div>
+					</button>
+					<span class="page-count">
+						Page {currentPage.toLocaleString()} / {pageCount.toLocaleString()}
+					</span>
+					<span class="print-page-count">
+						{data.length.toLocaleString()} of {totalRows.toLocaleString()} records
+					</span>
+					<button
+						aria-label="next-page"
+						class="page-changer"
+						class:hovering
+						disabled={currentPage === pageCount}
+						on:click={() => goToPage(currentPage + 1)}
+					>
+						<div class="page-icon">
+							<MdNavigateNext />
+						</div>
+					</button>
+					<button
+						aria-label="last-page"
+						class="page-changer"
+						class:hovering
+						disabled={currentPage === pageCount}
+						on:click={() => goToPage(pageCount)}
+					>
+						<div class="page-icon">
+							<MdLastPage />
+						</div>
+					</button>
+				</div>
+				<DownloadData class="download-button" {data} display={hovering} />
+			</div>
+		{:else}
+			<div class="table-footer">
+				<DownloadData class="download-button" {data} display={hovering} />
+			</div>
+		{/if}
+
+		<div class="noresults">No Results</div>
+	</div>
+{:else}
+	<ErrorChart {error} chartType="Data Table" />
+{/if}
+
+<style>
+	.table-container {
+		font-size: 9.5pt;
+		width: 97%;
+	}
```

Comment:
Is this file copy-pasted from somewhere? We should normalize as much of this into tailwind as possible at some point. This is also a really bulky component, so if we can break it up that would be good; I didn't see any obvious places to do that though.

Reply:
Yeah it's a slimmed down version of the `core-components` `DataTable`, so could probably share CSS between the two somehow

Reply:
Is it similar enough to have a shared `Table` component or something that both are leveraging? This plugin can rely on core-components safely, so it could be exported from there and ingested here

Reply:
There's some subtle differences that would make the script logic different but the markup might be able to be shared